### PR TITLE
Self on empty and better keySet logic

### PR DESCRIPTION
**Pull request by clicman**

### CODE REVIEW

File: datajack-api/src/test/java/ru/sbtqa/tag/datajack/callback/SampleDataGensCallback.java [hunk 1]
```diff
@@ -30,6 +30,7 @@ public Object call(CallbackData callbackData) {
             String genResult = null;
 
             try {
+                Thread.sleep(1);
```
Comment (kosteman) [inline datajack-api/src/test/java/ru/sbtqa/tag/datajack/callback/SampleDataGensCallback.java:33]: dafuq?

Comment (clicman) [reply datajack-api/src/test/java/ru/sbtqa/tag/datajack/callback/SampleDataGensCallback.java:33]: WAT?! It`s for different generation results (timestamp increment).

**Review by kosteman — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -162,20 +162,20 @@
...
-          <execution>
-            <id>sign-artifacts</id>
-            <phase>verify</phase>
-            <goals>
-              <goal>sign</goal>
-            </goals>
-          </execution>
-        </executions>
-      </plugin>
+      <!--<plugin>-->
```
Comment (PavelAlennikov) [inline pom.xml]: uncomment

File: adaptors/excel-adaptor/src/test/java/ru/sbtqa/tag/datajack/adaptors/ExcelDataTest.java [hunk 3]
```diff
@@ -178,13 +178,23 @@ public void toMapTest() throws DataException {
     @Test
     public void getKeySetTest() throws DataException {
         TestDataProvider tdo = new ExcelDataAdaptor(this.excellDataPath, collectionName);
-        Object supposedToBeSet = tdo.getKeySet();
+        Object supposedToBeSet = tdo.get("Common").getKeySet();
```
Comment (PavelAlennikov) [inline adaptors/excel-adaptor/src/test/java/ru/sbtqa/tag/datajack/adaptors/ExcelDataTest.java]: tdo -> testDataObject, which already became testDataProvider

File: adaptors/excel-adaptor/src/test/java/ru/sbtqa/tag/datajack/adaptors/ExcelDataTest.java [hunk 4]
```diff
@@ -178,13 +178,23 @@ public void toMapTest() throws DataException {
...
 
         assertTrue("Type of return value getKeySet() is not Set", supposedToBeSet instanceof Set);
         assertNotNull("Set object is null", supposedToBeSet != null);
         assertFalse("", ((Set) supposedToBeSet).isEmpty());
     }
 
+    @Test
+    public void emptyKeySetForValueTest() throws DataException {
+        TestDataProvider tdo = new ExcelDataAdaptor(this.excellDataPath, collectionName);
```
Comment (PavelAlennikov) [inline adaptors/excel-adaptor/src/test/java/ru/sbtqa/tag/datajack/adaptors/ExcelDataTest.java]: testDataProvider

File: adaptors/excel-adaptor/src/test/java/ru/sbtqa/tag/datajack/adaptors/ExcelDataTest.java [hunk 5]
```diff
@@ -178,13 +178,23 @@ public void toMapTest() throws DataException {
...
+        Object supposedToBeSet = tdo.get("Common.price").getKeySet();
+
+        assertTrue("Type of return value getKeySet() is not Set", supposedToBeSet instanceof Set);
+        assertNotNull("Set object is null", supposedToBeSet != null);
+        assertTrue("", ((Set) supposedToBeSet).isEmpty());
```
Comment (PavelAlennikov) [inline adaptors/excel-adaptor/src/test/java/ru/sbtqa/tag/datajack/adaptors/ExcelDataTest.java]: assert message is missng

File: adaptors/excel-adaptor/src/test/java/ru/sbtqa/tag/datajack/adaptors/ExcelDataTest.java [hunk 6]
```diff
@@ -195,6 +205,15 @@ public void getValuesTest() throws DataException {
         assertFalse("Collection of values is empty", ((Collection) rawValues).isEmpty());
     }
 
+    @Test
+    public void getEmptySelfTest() throws DataException {
+        TestDataProvider tdo = new ExcelDataAdaptor(this.excellDataPath, collectionName);
```
Comment (PavelAlennikov) [inline adaptors/excel-adaptor/src/test/java/ru/sbtqa/tag/datajack/adaptors/ExcelDataTest.java]: testDataProvider

File: datajack-api/src/main/java/ru/sbtqa/tag/datajack/adaptors/AbstractDataAdaptor.java [hunk 7]
```diff
@@ -1,12 +1,13 @@
...
 
 import com.mongodb.BasicDBObject;
+import ru.sbtqa.tag.datajack.TestDataProvider;
 import ru.sbtqa.tag.datajack.callback.GeneratorCallback;
 import ru.sbtqa.tag.datajack.exceptions.DataException;
 
 import java.util.*;
 
-public abstract class AbstractDataObjectAdaptor {
+public abstract class AbstractDataAdaptor implements TestDataProvider {
```
Comment (PavelAlennikov) [inline datajack-api/src/main/java/ru/sbtqa/tag/datajack/adaptors/AbstractDataAdaptor.java]: why are there Adaptor in every class name when we implement Provider one?

File: adaptors/json-adaptor/src/test/java/ru/sbtqa/tag/datajack/adaptors/json/JsonDataTest.java [hunk 8]
```diff
@@ -270,6 +270,25 @@ public void getKeySetTest() throws DataException {
...
     }
 
+    @Test
+    public void emptyKeySetForValueTest() throws DataException {
+        TestDataProvider tdo = new JsonDataAdaptor(JSON_DATA_PATH, "DataBlocks");
+        Object supposedToBeSet = tdo.get("Common.price").getKeySet();
+
+        assertTrue("Type of return value getKeySet() is not Set", supposedToBeSet instanceof Set);
+        assertNotNull("Set object is null", supposedToBeSet != null);
+        assertTrue("", ((Set) supposedToBeSet).isEmpty());
```
Comment (PavelAlennikov) [inline adaptors/json-adaptor/src/test/java/ru/sbtqa/tag/datajack/adaptors/json/JsonDataTest.java]: assert message

File: adaptors/mongo-adaptor/src/test/java/ru/sbtqa/tag/datajack/adaptors/MongoDataTest.java [hunk 9]
```diff
@@ -234,4 +236,47 @@ public void failRefAsObject() throws DataException {
...
+    }
+
+    @Test
+    public void getKeySetTest() throws DataException {
+        String collection = "DataBlocks";TestDataProvider tdo = new MongoDataAdaptor(mongoDb, collection);
+        Object supposedToBeSet = tdo.get("Common").getKeySet();
+
+        assertTrue("Type of return value getKeySet() is not Set", supposedToBeSet instanceof Set);
+        assertNotNull("Set object is null", supposedToBeSet != null);
+        assertFalse("", ((Set) supposedToBeSet).isEmpty());
```
Comment (PavelAlennikov) [inline adaptors/mongo-adaptor/src/test/java/ru/sbtqa/tag/datajack/adaptors/MongoDataTest.java]: message?

**Review by PavelAlennikov — CHANGES_REQUESTED** [review]

File: providers/excel-provider/src/main/java/ru/sbtqa/tag/datajack/providers/ExcelDataProvider.java [hunk 10]
```diff
@@ -84,7 +84,7 @@ public TestDataProvider get(String key) throws DataException {
         }
 
         this.way = key;
-        ExcelDataAdaptor tdo;
+        ExcelDataProvider tdo;
```
Comment (PavelAlennikov) [inline providers/excel-provider/src/main/java/ru/sbtqa/tag/datajack/providers/ExcelDataProvider.java]: tdo

File: providers/excel-provider/src/main/java/ru/sbtqa/tag/datajack/providers/ExcelDataProvider.java [hunk 11]
```diff
@@ -135,8 +135,8 @@ public TestDataProvider get(String key) throws DataException {
     }
 
     @Override
-    public ExcelDataAdaptor fromCollection(String collName) throws DataException {
-        ExcelDataAdaptor newObj = new ExcelDataAdaptor(this.dataFileName, this.workBook, collName);
+    public ExcelDataProvider fromCollection(String collName) throws DataException {
+        ExcelDataProvider newObj = new ExcelDataProvider(this.dataFileName, this.workBook, collName);
```
Comment (PavelAlennikov) [inline providers/excel-provider/src/main/java/ru/sbtqa/tag/datajack/providers/ExcelDataProvider.java]: what is newObj?

File: providers/excel-provider/src/test/java/ru/sbtqa/tag/datajack/providers/ExcelDataTest.java [hunk 12]
```diff
@@ -35,140 +35,140 @@ public void setUp() {
...
 
     @Test
     public void failWithCyclicReference() throws DataException, IOException, InvalidFormatException {
         String cyclicPath = "Common.cyclic";
-        TestDataProvider tdo = new ExcelDataAdaptor(this.excellDataPath, collectionName);
+        TestDataProvider testDataProvider = new ExcelDataProvider(this.excellDataPath, collectionName);
         String cyclicObject = "{ \"value\" : { \"sheetName\" : \"DataBlocks\", "
                 + "\"path\" : \"AnotherObject.cyclicRef\" }";
         expectDataExceptions
                 .expect(CyclicReferencesExeption.class);
```
Comment (PavelAlennikov) [inline providers/excel-provider/src/test/java/ru/sbtqa/tag/datajack/providers/ExcelDataTest.java:103]: format

File: providers/json-provider/src/main/java/ru/sbtqa/tag/datajack/providers/json/JsonDataProvider.java [hunk 13]
```diff
@@ -109,28 +109,28 @@ public static boolean isArray(String key) {
...
+        return (T) new JsonDataProvider(testDataFolder, obj, collectionName, extension);
     }
 
     @Override
-    public JsonDataAdaptor fromCollection(String collName) throws DataException {
+    public JsonDataProvider fromCollection(String collName) throws DataException {
         String json = readFile(this.testDataFolder, collName);
         BasicDBObject parsed = parse(json);
-        JsonDataAdaptor newObj = privateInit(this.testDataFolder, parsed, collName);
+        JsonDataProvider newObj = privateInit(this.testDataFolder, parsed, collName);
```
Comment (PavelAlennikov) [inline providers/json-provider/src/main/java/ru/sbtqa/tag/datajack/providers/json/JsonDataProvider.java]: newObj

File: providers/json-provider/src/main/java/ru/sbtqa/tag/datajack/providers/json/JsonDataProvider.java [hunk 14]
```diff
@@ -161,7 +161,7 @@ private TestDataProvider getSimple(String key) throws FieldNotFoundException, Da
         if (!(result instanceof BasicDBObject)) {
             result = new BasicDBObject(key, result);
         }
-        JsonDataAdaptor tdo = privateInit(this.testDataFolder, (BasicDBObject) result, this.collectionName, this.way);
+        JsonDataProvider tdo = privateInit(this.testDataFolder, (BasicDBObject) result, this.collectionName, this.way);
```
Comment (PavelAlennikov) [inline providers/json-provider/src/main/java/ru/sbtqa/tag/datajack/providers/json/JsonDataProvider.java]: tdo

File: providers/json-provider/src/main/java/ru/sbtqa/tag/datajack/providers/json/JsonDataProvider.java [hunk 15]
```diff
@@ -176,7 +176,7 @@ private TestDataProvider getSimple(String key) throws FieldNotFoundException, Da
 
     private TestDataProvider getComplex(String key) throws FieldNotFoundException, DataException {
 
-        JsonDataAdaptor tdo = privateInit(this.testDataFolder, parseComplexDBObject(key), this.collectionName, this.way);
+        JsonDataProvider tdo = privateInit(this.testDataFolder, parseComplexDBObject(key), this.collectionName, this.way);
```
Comment (PavelAlennikov) [inline providers/json-provider/src/main/java/ru/sbtqa/tag/datajack/providers/json/JsonDataProvider.java]: tdo

File: providers/mongo-provider/src/test/java/ru/sbtqa/tag/datajack/providers/MongoDataTest.java [hunk 16]
```diff
@@ -159,46 +159,46 @@ public void genDataSameCollection() throws DataException {
...
-        assertEquals(genGenOrgigin, tdo.get("Common").get("gen gen").get("gendata").getValue());
-        assertEquals(genGenOrgigin, tdo.get("Common.gen gen.gendata").getValue());
+        assertEquals(genGenOrgigin, testDataProvider.get("Common").get("gen gen").get("gendata").getValue());
+        assertEquals(genGenOrgigin, testDataProvider.get("Common.gen gen.gendata").getValue());
     }
 
     @Test
     public void genDataDifferentCollectionWithRefId() throws DataException {
-        TestDataProvider tdo = new MongoDataAdaptor(mongoDb, "DataBlocks").
+        TestDataProvider testDataProvider = new MongoDataProvider(mongoDb, "DataBlocks").
```
Comment (PavelAlennikov) [inline providers/mongo-provider/src/test/java/ru/sbtqa/tag/datajack/providers/MongoDataTest.java:187]: format

File: providers/properties-provider/src/test/java/ru/sbtqa/tag/datajack/providers/properties/PropertiesDataTest.java [hunk 17]
```diff
@@ -264,28 +264,28 @@ public void toMapTest() throws DataException {
...
     @Test
     public void emptyKeySetForValueTest() throws DataException {
-        TestDataProvider tdo = new PropertiesDataAdaptor(this.propertiesDataPath, "DataBlocks");
+        TestDataProvider tdo = new PropertiesDataProvider(this.propertiesDataPath, "DataBlocks");
         Object supposedToBeSet = tdo.get("Common.price").getKeySet();
 
         assertTrue("Type of return value getKeySet() is not Set", supposedToBeSet instanceof Set);
         assertNotNull("Set object is null", supposedToBeSet != null);
-        assertTrue("", ((Set) supposedToBeSet).isEmpty());
+        assertTrue("testDataProvider", ((Set) supposedToBeSet).isEmpty());
```
Comment (PavelAlennikov) [inline providers/properties-provider/src/test/java/ru/sbtqa/tag/datajack/providers/properties/PropertiesDataTest.java]: wrong message

**Review by PavelAlennikov — CHANGES_REQUESTED** [review]

File: providers/properties-provider/src/test/java/ru/sbtqa/tag/datajack/providers/properties/PropertiesDataTest.java [hunk 18]
```diff
@@ -264,17 +264,38 @@ public void toMapTest() throws DataException {
...
 
+
+    @Test
+    public void emptyKeySetForValueTest() throws DataException {
+        TestDataProvider tdo = new PropertiesDataProvider(this.propertiesDataPath, "DataBlocks");
+        Object supposedToBeSet = tdo.get("Common.price").getKeySet();
+
+        assertTrue("Type of return value getKeySet() is not Set", supposedToBeSet instanceof Set);
+        assertNotNull("Set object is null", supposedToBeSet != null);
+        assertTrue("testDataProvider", ((Set) supposedToBeSet).isEmpty());
```
Comment (PavelAlennikov) [inline providers/properties-provider/src/test/java/ru/sbtqa/tag/datajack/providers/properties/PropertiesDataTest.java]: wrong message

**Review by PavelAlennikov — CHANGES_REQUESTED** [review]

File: providers/mongo-provider/src/test/java/ru/sbtqa/tag/datajack/providers/MongoDataTest.java (hunk 16 above)
Comment (clicman) [reply providers/mongo-provider/src/test/java/ru/sbtqa/tag/datajack/providers/MongoDataTest.java:187]: format?

**Review by PavelAlennikov — APPROVED** [review]

**Pull request merged**